### ISSUE TITLE
Desktop: Some Window methods may not be used from Scroll.OnScrolled/Layout.Layout: freeze/deadlock

### ISSUE THREAD

**Issue by pjanx** [fyne-io/fyne#5266]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

It is not directly possible to use all Window methods in certain contexts, such as from Scroll.OnScrolled, or Layout.Layout.

The straight-forward workaround is to run them from another goroutine, so that it can wait on the lock asynchronously.

The questions are:
 - Is this non-recursiveness documented anywhere?
 - Should it work, possibly by actually queuing such operations? (The funcQueue variable, as I understand it, is a misnomer, because the order is undefined with unbuffered channels.)
 - Should it rather detect such situations and panic?

### How to reproduce

* Start the application
* Scroll the window
* It freezes [here](https://github.com/fyne-io/fyne/blob/81b3cd1496f0a41dc8dee09f8f60722575274824/internal/driver/glfw/loop.go#L52) and stops responding

### Example code

```
package main

import (
	"bytes"
	"image"
	"image/png"
	"strings"

	"fyne.io/fyne/v2"
	"fyne.io/fyne/v2/app"
	"fyne.io/fyne/v2/container"
	"fyne.io/fyne/v2/widget"
)

func main() {
	ib := bytes.NewBuffer(nil)
	_ = png.Encode(ib, image.NewRGBA(image.Rect(0, 0, 48, 48)))
	ir := fyne.NewStaticResource("Icon.png", ib.Bytes())

	a := app.New()
	w := a.NewWindow("Bugs")
	rt := widget.NewRichText()
	rt.Wrapping = fyne.TextWrapWord
	rt.ParseMarkdown(strings.Repeat("Lorem ipsum dolor sit amet\n\n", 100))
	rs := container.NewVScroll(rt)
	// Or: w.SetFullScreen(true)
	// Or: w.CenterOnScreen()
	rs.OnScrolled = func(fyne.Position) { w.SetIcon(ir) }
	w.SetContent(rs)
	w.ShowAndRun()
}
```

### Fyne version

2.5.2

### Go compiler version

1.23.2

### Operating system and version

Linux

**Comment by pjanx** [comment]
Duplicate of #4155
